### PR TITLE
D language track

### PR DESCRIPTION
**Pull request by celavek**
I have implemented the first 10 problems for the the D language track in order to be able to enable it on the sit. More problems are to come.

I have not tested the track in the full sit environment as I do not know how to do that exactly.
I tried following the documentation but it is kind of confusing and it feels dispersed - I do not have much prior exposure to the technologies used. I am willing to do that(test the track on the site/problem engine) if someone could provide me with some clear steps on how to proceed.

Thanks

### CODE REVIEW

**Comment by celavek** [comment]
Is anybody interested in this? Any sort of answer will do.

**Comment by kytrinyx** [comment]
Hi. I'm so sorry it's taken a week to get back to you!

This is EPIC.

> I tried following the documentation but it is kind of confusing and it feels dispersed

Yeah, I'm so sorry about that. I'm working on rewriting the documentation, but it's still kind of a mess.

I've updated the launch checklist issue: https://github.com/exercism/xdlang/issues/6

I'll review this immediately, and then we can chat about how to test.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,7 @@
 bin/configlet
 bin/configlet.exe
+*.swp
+*.swo
```
Comment (kytrinyx) [inline .gitignore]: Would you add the swap files to your global gitignore file?

https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

File: config.json [hunk 2]
```diff
@@ -3,15 +3,24 @@
   "language": "D",
   "repository": "https://github.com/exercism/xdlang",
   "checklist_issue": 6,
-  "active": false,
+  "active": true,
```
Comment (kytrinyx) [inline config.json]: I'd like to activate the track in a separate pull request.

File: docs/LEARNING.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+## Learning D
```
Comment (kytrinyx) [inline docs/LEARNING.md]: We don't need titles in these doc files; they get included into pages that we already have titles in.

See https://github.com/exercism/xgo/tree/master/docs for reference.

File: docs/LEARNING.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+better programmers.
+
+Here are a couple of resources to get you started:
+
+* [Learning D as a beginner](http://ddili.org/ders/d.en/index.html), Ali Cehreli excellent book
+* [Learning D for experienced programmers](http://wiki.dlang.org/Coming_From) coming from other languages
+* [D Tutorials](https://wiki.dlang.org/Tutorials)
+* [Overview of D's unique features](http://dlang.org/overview.html)
+* [Common questions about D](http://dlang.org/faq.html)
+* [DConf](http://dconf.org/2016/index.html) presentations and videos on YouTube
```
Comment (kytrinyx) [inline docs/LEARNING.md]: I love that you included conference videos!

**Comment by kytrinyx** [comment]
I have a few comments—nothing major. This looks awesome! I'll be back in 30 minutes with some stuff about testing the exercises within Exercism.

**Comment by kytrinyx** [comment]
OK, I'm back. I'm going to tweak this a bit locally and then merge, and then write up something about how to test.

**Comment by kytrinyx** [comment]
I merged this in c0012db69396eb2b1c223575df038e85a8bbde8e

**Comment by kytrinyx** [comment]
I've updated the API to the latest version, which includes these problems and the documentation.

You can verify the interaction with the command-line client with:

```
$ exercism fetch dlang
```

or

```
$ exercism fetch dlang series # to fetch a specific exercise
```

The documentation is alive at http://exercism.io/languages/dlang

I'll give you mentorship status on the track so you see any submissions that come in (and commit to this repository).

Thank you so much! :heart:

**Comment by celavek** [comment]
Wow! That was fast. Thank you!
I have tested the track (fetching and submitting) and it works great.
Do I have anything to do regarding or as a side effect of this pull-request - I see it's closed 
but there were some comments from you.

**Comment by kytrinyx** [comment]
> Do I have anything to do regarding or as a side effect of this pull-request - I see it's closed 
> but there were some comments from you.

Take a look through the documentation and check if it looks OK the way it is. We may have ended up with a couple places where we have two headlines in a row. If we do, then it should be an easy fix.

**Pull request closed without merging**